src/Core/Framework/Api/Controller/AuthController.php

```diff
@@ -49,7 +50,11 @@ public function token(Request $request): Response
         $psr7Request = $this->psrHttpFactory->createRequest($request);
         $psr7Response = $this->psrHttpFactory->createResponse($response);
 
-        $response = $this->authorizationServer->respondToAccessTokenRequest($psr7Request, $psr7Response);
+        try {
+            $response = $this->authorizationServer->respondToAccessTokenRequest($psr7Request, $psr7Response);
+        } catch (OAuthServerException $e) {
+            throw new OAuthServerException($e->getMessage(), $e->getCode(), $e->getErrorType(), 401);
```

Comment:
FYI: This seems a bit weird, to throw another exception of the same class, but it's necessary because the [login JS](https://github.com/shopware/shopware/blob/trunk/src/Administration/Resources/app/administration/src/module/sw-login/view/sw-login-login/index.js#L52-L57) is depending on it.